packages/react-liveness/src/components/FaceLivenessDetector/service/machine/index.ts

```diff
@@ -447,6 +458,22 @@ export const livenessMachine = createMachine<LivenessContext, LivenessEvent>(
           freshnessColorEl: event.data?.freshnessColorEl,
         }),
       }),
+      updateDeviceAndStream: assign({
+        videoAssociatedParams: (context, event) => {
+          localStorage.setItem(CAMERA_ID_KEY, event.data?.newDeviceId);
```

Comment:
This feels like a side effect?

Reply:
Can be removed, was just added as a way to persist camera preference

Reply:
I dont think it can be removed? We do need to persist the camera preference.

Reply:
Persisting the camera preference is fine, but usage of side effects in the state machine removes determinism from the machine itself